src/mlir/generator.h

```diff
@@ -44,6 +44,12 @@ namespace mlir::verona
     /// Symbol tables for variables.
     SymbolTableT symbolTable;
 
+    /// Set of known arithmetic operations and their number of operators
+    std::map<llvm::StringRef, size_t> arithmetic;
```

Comment:
This probably wants to be a `std::unordered_list` (lookup is faster and I don't think you're using the ordering).  [`llvm::StringMap`](https://llvm.org/doxygen/classllvm_1_1StringMap.html) may be even better.

Reply:
A `StringSwitch` like you proposed below would be even better.